src/OrbitBase/File.cpp

```diff
@@ -197,9 +197,15 @@ ErrorMessageOr<size_t> ReadFullyAtOffset(const unique_fd& fd, void* buffer, size
                                          int64_t offset) {
   uint8_t* current_position = static_cast<uint8_t*>(buffer);
   size_t bytes_read = 0;
-  while (bytes_read < size) {
+  while (size > 0) {
     int64_t pread_result = TEMP_FAILURE_RETRY(pread(fd.get(), current_position, size, offset));
     if (pread_result == -1) {
+      // When pread fails on a subsequent call we will return what we have read so far.
+      // This is not uncommon when trying to read over the end of a file. The first call
+      // will return less bytes then requested. The next call will fail because the offset
+      // will be invalid.
+      if (bytes_read > 0) return bytes_read;
```

Comment:
I'm not sure if that's our desired behaviour. At least it's expected to behave like so by the UserSpaceInstrumentationTests.

There is one alternative: Instead I could check the value of `pread_result` and see if it is smaller than `size` but not equal to `0x7ffff000` and break the loop in that case. Could be better in some ways, but I don't like the dependency on that arbitrary constant. On the other hand this constant will probably never change again.

Reply:
Let me know what you think.

Reply:
While the test is expecting this behaviour (AccessTraceesMemoryTest.cpp:107) I do not really need this. The places I use `ReadFullyAtOffset `in real code are guaranteed to not step over the end of the file. So unconditionally returning an error in the  `(pread_result == -1) `case would be fine as well.

Let's chat briefly tomorrow. I am not sure that pread_result == -1 && bytes_read > 0 always indicates a success. So returning an error might be the better solution.

Reply:
I changed the implementation to use a seek and `FullyRead` instead of `pread`. That also simplifies the implementation quite a bit. Let me know waht you think